lightning/src/ln/channelmanager.rs

```diff
@@ -6600,49 +6602,48 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> Writeable f
 	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
 		let _consistency_lock = self.total_consistency_lock.write().unwrap();
 
+		let forward_htlcs = self.forward_htlcs.lock().unwrap();
+		let channel_state = self.channel_state.lock().unwrap();
+		let per_peer_state = self.per_peer_state.write().unwrap();
+		let peer_states = per_peer_state.iter().map(|(peer_pubkey, peer_state_mutex)| (peer_pubkey, peer_state_mutex.lock().unwrap())).collect::<Vec<(&PublicKey, MutexGuard<'_, PeerState>)>>();
```

Comment:
@TheBlueMatt are we okay this creating a lock order for all the inner locks of the `PeerState` mutexes? In the "non-serialization" state (ie. outside of `ChannelManager::write`) we should never acquire two `PeerState` locks at the same time, due to the deadlock possibility. Though I'm trying to figure out what consequences this will have when we move towards parallelization..

If we are not ok with this, I don't think we can take the locks in the beginning of the `ChannelManager::write` function, and revert back to how this PR originally acquired the locks.

Reply:
Ok :)! I'll go ahead and remove the commit that takes all the locks in the beginning of the function.